tests/integration/utils.py

```diff
@@ -93,6 +93,29 @@ def wait_for_complete_request(self, response):
 
         return response
 
+    def fetch_all_requests(self, request_json=None):
+        """
+        Fetch a list of requests from the Cachito API.
+
+        :param dict request_json: Request parameters and values (page, per_page, status, verbose)
+        :return: Object that contains response from the Cachito API
+        :rtype: list
+        """
+        if not request_json:
+            request_json = {}
+        url_values = urllib.parse.urlencode(request_json)
+        request_url = f"{self._cachito_api_url}/requests?{url_values}"
+        all_items = []
+        while request_url:
+            resp = requests.get(request_url, qtimeout=15)
```

Comment:
You can let the requests library add the query parameters:
https://requests.readthedocs.io/en/master/user/quickstart/#passing-parameters-in-urls

```suggestion
        request_url = f"{self._cachito_api_url}/requests"
        all_items = []
        while request_url:
            resp = requests.get(request_url, params=request_json, timeout=15)
```

Reply:
There was also a typo in the `timeout` keyword argument

Reply:
Done, sorry for the typo.